actions/llm-action-error-reporter/action.yml

```diff
@@ -53,39 +82,49 @@ runs:
 
         # Convert pull_requests array to JSON string and get pr_number from the first object
         pull_requests_json='${{ toJson(github.event.workflow_run.pull_requests) }}'
-        echo "pr_number="$(echo "$pull_requests_json" | jq '.[0].number')"" >> "$GITHUB_OUTPUT"
-
-    - name: Prepare logs
-      if:
-        ${{ env.SKIP_ACTION == 'false' && inputs.parent-workflow-conclusion ==
-        'failure' }}
-      shell: bash
-      env:
-        GH_TOKEN: ${{ inputs.gh-token }}
-      run: |
-        echo "Analyzing run ${{ github.event.workflow_run.id }}"
-        gh run view ${{ github.event.workflow_run.id }} --log-failed | jq -Rs . | sed 's/^"//;s/"$//' > runlog.log
+        pr_number=$(echo "$pull_requests_json" | jq -r '.[0].number')
+        echo "pr_number=$pr_number" >> "$GITHUB_OUTPUT"
 
     - name: Checkout action repo
       if: ${{ env.SKIP_ACTION == 'false' }}
       uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
       with:
         repository: smartcontractkit/.github
         ref: ${{ inputs.workflow-ref }}
-        path: "error_reporter_actions"
-        # prevents access to any other action and workflow files
+        path:
+          "error_reporter_actions"
+          # prevents access to any other action and workflow files
         sparse-checkout: |
           actions/llm-action-error-reporter/log-analyze-prompt.md
+          actions/llm-action-error-reporter/update_pr_comment.sh
+          actions/llm-action-error-reporter/trim_logs.sh
         sparse-checkout-cone-mode: false
         fetch-depth: 1
 
+    - name: Prepare logs
+      if:
+        ${{ env.SKIP_ACTION == 'false' && inputs.parent-workflow-conclusion ==
+        'failure' }}
+      shell: bash
+      env:
+        GH_TOKEN: ${{ inputs.gh-token }}
+        LINE_LIMIT: ${{ inputs.log-lines-limit }}
+      run: |
+        echo "Analyzing run ${{ github.event.workflow_run.id }}"
+        LOG_RAW=$(gh run view ${{ github.event.workflow_run.id }} --log-failed)
+        echo "$LOG_RAW" > rawlog.log
+
+        # trim logs and outputs to runlog.log
+        chmod +x ./error_reporter_actions/actions/llm-action-error-reporter/trim_logs.sh
+        ./error_reporter_actions/actions/llm-action-error-reporter/trim_logs.sh
+
     - name: Call ChatGPT
       if:
         ${{ env.SKIP_ACTION == 'false' && inputs.parent-workflow-conclusion ==
         'failure' }}
       shell: bash
       run: |
-        log_output=$(cat runlog.log | sed -e 's/\\ No newline at end of file//g' | sed -E 's/\\+$//g' | sed -E 's/\\+ //g')
+        log_output=$(cat runlog.log | jq -Rs . | sed 's/^"//;s/"$//' | sed -e 's/\\ No newline at end of file//g' | sed -E 's/\\+$//g' | sed -E 's/\\+ //g')
```

Comment:
It's not immediately obvious to me what these sed's are doing. Would be great to add a comment here to explain.